test/ecto/adapters/postgres_test.exs

```diff
@@ -545,6 +545,109 @@ defmodule Ecto.Adapters.PostgresTest do
+    query = Schema
+            |> select([r], r.x)
+            |> window(w as ^partition)
+            |> plan()
+    assert all(query) == ~s{SELECT s0."x" FROM "schema" AS s0 WINDOW w AS (PARTITION BY s0."x")}
+  end
+
+  test "two windows" do
+    partition_by_y = partition_by([r], r.y)
+    query = Schema
+            |> select([r], r.x)
+            |> window([r], w1 as partition_by r.x)
+            |> window(w2 as ^partition_by_y)
+            |> plan()
+    assert all(query) == ~s{SELECT s0."x" FROM "schema" AS s0 WINDOW w1 AS (PARTITION BY s0."x"), w2 AS (PARTITION BY s0."y")}
+  end
+
+  test "two windows with the same name" do
+    query = Schema
+            |> select([r], r.x)
+            |> window([r], w as partition_by r.x)
+
+    # FIXME: move to the builder specs
+    assert_raise Ecto.Query.CompileError, fn ->
+      query |> window([r], w as partition_by r.y)
+    end
+  end
+
+  test "multifiled partition by" do
+    query = Schema
+            |> window([r], w as partition_by([r.x, r.z]))
+            |> select([r], r.x)
+            |> plan()
+    assert all(query) == ~s{SELECT s0."x" FROM "schema" AS s0 WINDOW w AS (PARTITION BY s0."x", s0."z")}
+  end
+
+  test "count over all" do
+    query = Schema
+            |> select([r], count(r.x) |> over)
+            |> plan()
+    assert all(query) == ~s{SELECT count(s0."x") OVER () FROM "schema" AS s0}
+  end
+
+  test "row_number over all" do
+    query = Schema
+            |> select([r], row_number(r.x) |> over)
+            |> plan()
+    assert all(query) == ~s{SELECT row_number(s0."x") OVER () FROM "schema" AS s0}
+  end
+
+  test "count over partition by (variable)" do
+    partition = partition_by([r], r.x)
+    query = Schema
+            |> select([r], count(r.x) |> over(^partition))
+            |> plan()
+    assert all(query) == ~s{SELECT count(s0."x") OVER (PARTITION BY s0."x") FROM "schema" AS s0}
+  end
+
+  test "count over partition by (inline)" do
+    query = Schema
+            |> select([r], count(r.x) |> over(partition_by [r.x, r.z]))
+            |> plan()
+    assert all(query) == ~s{SELECT count(s0."x") OVER (PARTITION BY s0."x", s0."z") FROM "schema" AS s0}
+  end
+
+  test "count over window" do
+    query = Schema
+            |> window([r], w as partition_by r.x)
+            |> select([r], count(r.x) |> over(w))
+            |> plan()
+    assert all(query) == ~s{SELECT count(s0."x") OVER w FROM "schema" AS s0 WINDOW w AS (PARTITION BY s0."x")}
+  end
+
+  test "count over ordered window (keywords)" do
+    query = from s in Schema,
+              window: (w as partition_by s.x, order_by: [s.a, s.b]),
+              select: (count(s.x) |> over(w))
+    query = query |> plan()
+    assert all(query) == ~s{SELECT count(s0."x") OVER w FROM "schema" AS s0 WINDOW w AS (PARTITION BY s0."x" ORDER BY s0."a", s0."b")}
+  end
+
+  test "count over unknown window" do
+    query = Schema
+            |> window([r], w as partition_by r.x)
```

Comment:
The use `as` as operator here is really smart but a bit foreign here since it is not an actual operator. My suggestion is the following:

```elixir
windows(r, w: partition_by(r.x))
```

i.e. rename the macro to `windows` and allow it to be a keyword list of windows names and their definitions. Similar to how `order_by` itself works. Thoughts?

Reply:
I tried to find a clear syntax for `WINDOW w AS …`, but I missed this obvious variant. Thank you!